README.md

```diff
@@ -93,6 +94,42 @@ and build the library with:
 npm run build-lib
 ```
 
+### Scripts For Development
+
+The following scripts can be used for development:
+
+- `npm test`: runs the unit tests of the demo application and the lib's unit tests.
+- `npm test-lib`: runs the lib's unit tests in headless mode.
+- `npm test-lib-local`: runs the lib's unit test. 
+- `npm run build`: builds the lib in productive mode.
+- `npm run build-dev`: builds the lib in development mode.
+- `npm build-app`: builds the demo application in productive mode.
+- `npm run yalc-publish-lib`: builds the lib and publishes it locally using `yalc`.
+- `npm run webdriver-update`: installs Chrome webdriver in the required version for the E2E test.
+- `npm run e2e`: runs the E2E tests from the demo application.
+
+#### Running Tests
+
+The library contains only unit tests. They can be run with `npm test-lib-local`
+and do not require a running instance of Knora.
+Note that `npm test` runs both the lib's and the demo application's unit test.
+
+
+The E2E tests have to be run from the demo application.
+They require a running instance of DSP-API compatible with the release specified in [vars.mk](vars.mk).
+Before running the E2E tests, reload the data in DSP-API's db.
+The E2E test can be run as follows:
+- `npm run webdriver-update`
+- `npm run e2e`
+
+#### Building the Library
+
+The library is built in productive mode for publication, using [tsconfig.lib.prod.json](projects/dsp-ui/tsconfig.lib.prod.json).
+
+To perform additional checks during development, the library can be built in development mode,
+using [tsconfig.lib.json](projects/dsp-ui/tsconfig.lib.json). Development builds use Ivy and perform [strict template checks](https://angular.io/guide/template-typecheck#template-type-checking).
+
+
 ### Contribution
 
 If you want to add more components, services and so on to a module of the library, you can do it with:
```

Comment:
Can we also add some details about the code conventions here? So that if anyone wants to contribute, they can follow the same code conventions.

Reply:
I think code conventions should go into the main contribution documentation because it will be the same for DSP-UI as well for DSP-App and maybe also for DSP-JS. Here we should add a link to the corresponding documentation (which doesn't exist yet): e.g. <https://docs.dasch.swiss/developers/dsp/frontend-code-conventions/>
@waychal It's a good point. We have to move the code conventions from the google docs to dsp-docs.